Added iterative solvers

phg1024 opened the pull request:
Conjugate gradient and BiCGSTAB are implemented.

phg1024 committed:
Added conjugate gradient solver. Supports both full matrix and sparse matrix.

phg1024 committed:
added bicgstab algorithm

phg1024 committed:
Merge branch 'master' of https://github.com/phg1024/numeric

phg1024 committed:
added test cases for iterative solvers.

phg1024 commented:
Added benchmark3 to show the iterative solvers.

![perf](https://f.cloud.github.com/assets/2273868/1496620/7597e9ac-47fd-11e3-899d-ec77a12dee21.PNG)

phg1024 committed:
1. minor change in the solvers
2. bug fixed in the benchmark page

sloisel commented:
This looks good. I want to merge it but I have to fix my build machine
first.

Thanks a lot for this. I'd write more but I'm on the phone.
On Nov 7, 2013 9:33 PM, "Peihong" notifications@github.com wrote:

> Added benchmark3 to show the iterative solvers.
> 
> [image: perf]https://f.cloud.github.com/assets/2273868/1496009/19fb692e-47f4-11e3-9363-7810c61ee0e2.PNG
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/sloisel/numeric/pull/49#issuecomment-28008587
> .

phg1024 committed:
added SOR solver.
the performance of SOR solver is not as good as CG and BiCGSTAB, but similar to ccsLUPSolve.
in practice, CG and BiCGSTAB should always be the best options.
